### PR TITLE
Add node pool label

### PR DESCRIPTION
**Pull request by shayorshay**
This PR adds a configurable node pool label in the pipeline specification. This is needed because the default label name differs per cloud provider.

### CODE REVIEW

File: src/fondant/pipeline.py [hunk 1]
```diff
@@ -80,6 +82,7 @@ def __init__(
         self.arguments.setdefault("component_spec", self.component_spec.specification)
 
         self.number_of_gpus = number_of_gpus
+        self.node_pool_label = node_pool_label
         self.node_pool_name = node_pool_name
         self.p_volumes = p_volumes
         self.ephemeral_storage_size = ephemeral_storage_size
```
Comment (PhilippeMoussalli) [inline src/fondant/pipeline.py:88]: Since both the label and the name depend on each other. Can you raise an error if one of them is not provided? 

```python
from fondant.exceptions import InvalidPipelineDefinition 
if bool(node_pool_label) != bool(node_pool_name):
        raise InvalidPipelineDefinition("Both 'node_pool_label' and 'node_pool_name' must be defined or both must be None.")
```

And can you also test that a valid error is returned by adding two example to `test_component_op`, one where the name is defined and the nodepool is not and the other way around: 

https://github.com/ml6team/fondant/blob/5b123bff57c933e3fbc24aac7a36f73bf4ddacd3/tests/test_pipeline.py#L35

**Review by PhilippeMoussalli — COMMENTED** [review]
Thanks @shayorshay ! :) 
This will come in handy for different clouds. Left one small comments to handle edge cases and maybe add a test for it.

File: src/fondant/pipeline.py (hunk 1 above)
Comment (shayorshay) [reply src/fondant/pipeline.py:90]: I added the error checking and test. Could you please check? @PhilippeMoussalli

Comment (PhilippeMoussalli) [reply src/fondant/pipeline.py:90]: Thanks :)

**Review by PhilippeMoussalli — APPROVED** [review]
